File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> m;
    for(int i = 0; i < clothes.size(); ++i)
    {
        ++m[clothes[i][1]];
    }
    
    for(auto count : m)
    {
        answer *= (count.second + 1);
    }
    
    answer -= 1;
    
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> clothCount = new HashMap<>();
        
        for (String[] cloth : clothes) {
            clothCount.put(cloth[1], clothCount.getOrDefault(cloth[1], 0) + 1);
        }
        
        int answer = 1;
        List<Integer> count = new ArrayList<>(clothCount.values());
        for (int c : count) {
            answer *= (c + 1);
        }
        
        return answer - 1;
    }
}